src/hooks/useDragAndDropThought.tsx

```diff
@@ -0,0 +1,230 @@
+    store.dispatch(
+      error({ value: `Cannot move the ${isRoot(thoughtsFrom) ? 'home' : 'em'} context to another context.` }),
+    )
+    return
+  }
+
+  // drop on itself or after itself is a noop
+  if (equalPath(thoughtsFrom, thoughtsTo) || isBefore(state, thoughtsFrom, thoughtsTo)) return
+
+  const parent = unroot(rootedParentOf(state, thoughtsTo))
+  const newPath = appendToPath(parent, head(thoughtsFrom))
+
+  const newRank = getRankBefore(state, thoughtsTo)
+  store.dispatch(
+    props.showContexts
+      ? createThought({
+          value: toThought.value,
+          path: thoughtsFrom,
+          rank: getNextRank(state, head(thoughtsFrom)),
+        })
+      : moveThought({
+          oldPath: thoughtsFrom,
+          newPath,
+          newRank,
+        }),
+  )
+
+  // alert user of move to another context
+  if (!sameParent) {
+    // wait until after MultiGesture has cleared the error so this alert does not get cleared
+    setTimeout(() => {
+      store.dispatch((dispatch, getState) => {
+        const state = getState()
+        const parentThought = getThoughtById(state, head(parentOf(thoughtsTo)))
+        if (!parentThought) return
+        const alertFrom = '"' + ellipsize(fromThought.value) + '"'
+        const alertTo = isRoot([parentThought.id]) ? 'home' : '"' + ellipsize(parentThought.value) + '"'
+        dispatch(
+          alert(`${alertFrom} moved to ${alertTo} context.`, {
+            alertType: AlertType.ThoughtMoved,
+            clearDelay: 5000,
+          }),
+        )
+      })
+    }, 100)
+  }
+}
+
+/** Collects props from the DragSource. */
+const dragCollect = (monitor: DragSourceMonitor) => ({
+  isDragging: monitor.isDragging(),
+})
+
+/** Collects props from the DropTarget. */
+const dropCollect = (monitor: DropTargetMonitor) => ({
+  isHovering: monitor.isOver({ shallow: true }) && monitor.canDrop(),
+  // is being hovered over current thought irrespective of whether the given item is droppable
+  isBeingHoveredOver: monitor.isOver({ shallow: true }),
+  isDeepHovering: monitor.isOver(),
+})
+
+/** A draggable and droppable Thought hook. */
+const useDragAndDropThought = (props: Partial<ThoughtContainerProps>) => {
+  const propsTypes = props as ThoughtContainerProps
+
+  const [{ isDragging }, drag, dragPreview] = useDrag({
+    item: { path: props.path, simplePath: props.simplePath, zone: DragThoughtZone.Thoughts, type: 'thought' },
+    begin: () => beginDrag(propsTypes),
+    canDrag: () => canDrag(propsTypes),
+    end: () => endDrag(),
+    collect: dragCollect,
+  })
+
+  const [{ isHovering, isBeingHoveredOver, isDeepHovering }, dropRef] = useDrop({
+    accept: ['thought', NativeTypes.FILE],
+    canDrop: (item, monitor) => canDrop(propsTypes, monitor),
+    drop: (item, monitor) => drop(propsTypes, monitor),
+    collect: dropCollect,
+  })
+
+  return { isDragging, drag, dragPreview, isHovering, isBeingHoveredOver, isDeepHovering, drop: dropRef }
```

Comment:
Pass `dropRef` as-is, or rename to `dropTarget`, to avoid confusion with the existing `drop` function.